### PR TITLE
Add Largest Sum Contiguous Subarray in CSharp

### PR DESCRIPTION
**Pull request by EmanuelTinoco**
**Fixes [#726](https://github.com/FOSS-UCSC/FOSSALGO/issues/726)**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-lscsa/csharp/LargestSumContiguousSubArray.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                maxEnd = 0;
+            }
+        }
+
+        return max;
+    }
+
+    public static void Main()
+    {
+        int[] arr = new int[] {-2, 1, -3, 4, -1, 2, 1, -5, 4};
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/csharp/LargestSumContiguousSubArray.cs:27]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6390379)

File: algorithms/ar-lscsa/csharp/LargestSumContiguousSubArray.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        int result = DoLargestSumContiguousSubArray(arr);
+
+        Console.WriteLine(result);
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/csharp/LargestSumContiguousSubArray.cs:30]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6390379)

File: algorithms/ar-lscsa/csharp/LargestSumContiguousSubArray.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+using System;
+
+public class LargestSumContiguousSubArray
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/csharp/LargestSumContiguousSubArray.cs:3]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6390379)

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
@all-contributors please add @david-azevedo for code and doc

**Comment by Danushka96** [comment]
@all-contributors please add @EmanuelTinoco  for code and doc

**Comment by Danushka96** [comment]
Hi @EmanuelTinoco 
Thanks for your contribution! keep it up!

**Pull request merged**